docker-plugin/src/test/java/com/nirima/jenkins/plugins/docker/DockerImagePullStrategyTest.java

```diff
@@ -0,0 +1,106 @@
+package com.nirima.jenkins.plugins.docker;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.ListImagesCmd;
+import com.github.dockerjava.api.model.Image;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/**
+ * @author Slawomir Jaranowski
+ */
+@RunWith(Parameterized.class)
+public class DockerImagePullStrategyTest {
+
+    private final List<Image> imageList;
+    private final String imageName;
+    private final DockerImagePullStrategy pullStrategy;
+    private final boolean shouldPull;
+
+    @Parameterized.Parameters(name = "existing image: ''{0}'', image to pull: ''{1}'', strategy: ''{2}''")
+    public static Collection<Object[]> data() {
```

Comment:
I initially wrote this strategies, can i borrow your fix & test into https://github.com/KostyaSha/yet-another-docker-plugin ?